lib/elixir/lib/calendar/datetime.ex

```diff
@@ -559,38 +560,48 @@ defmodule DateTime do
   end
 
   @doc """
-  Converts a DateTime from one calendar to another.
+  Converts a `DateTime` from one calendar to another.
 
-  If this conversion fails for some reason, an `{:error, reason}` tuple is returned.
+  If it is not possible to convert unambiguously between the calendars
+  (see `Calendar.compatible_calendars?/2`), an `{:error, :incompatible_calendars}` tuple
+  is returned.
 
   ## Examples
 
+  Imagine someone implements `Calendar.Julian`:
```

Comment:
Based on https://github.com/elixir-lang/elixir/pull/5921#issuecomment-289207070. Happy to prepare a PR for other types.